samples/gatsby-preview/src/index.jsx

```diff
@@ -0,0 +1,165 @@
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    let parameters = this.props.sdk.parameters
+    let projectId = parameters.installation.projectId
+    let contentTypeSlug = parameters.instance.contentTypeSlug
+    let isAutoUpdate = parameters.instance.isAutoUpdate
+
+    this.state = {
+      projectId: projectId,
+      projectUrl:
+        "https://" + projectId + "-preview.gtsb.io/" + contentTypeSlug + "/",
+      isAutoUpdate: isAutoUpdate,
+      webhookUrl:
+        "https://backends.ctffns.net/gatsby-preview-proxy/" + projectId,
+      debounceInterval: null
+    }
+  }
+
+  componentDidMount = () => {
+    this.detachFns = []
+
+    this.detachFns.push(this.props.sdk.entry.onSysChanged(this.onSysChanged))
```

Comment:
No need for a list of listeners, just do:

`this.off = this.props.sdk.entry.onSysChanged(this.onSysChanged)`

and

`this.off()`

in `componentWillUnmount`